rpc/http.go

```diff
@@ -65,8 +71,9 @@ func (hc *httpConn) Close() error {
 	return nil
 }
 
-// DialHTTP creates a new RPC clients that connection to an RPC server over HTTP.
-func DialHTTP(endpoint string) (*Client, error) {
+// DialHTTPClient creates a new RPC clients that connection to an RPC server over HTTP
+// using the provided HTTP Client.
+func DialHTTPClient(endpoint string, client HttpClient) (*Client, error) {
```

Comment:
The usual "way" in Go code when initializing something with an extra parameter is `With`. I.e. `DialHTTPWithClient`. Please rename.

Reply:
Yeah that makes sense.